### ISSUE TITLE
Build failure due to change of LLVM function signature: createMCCodeEmitter

### ISSUE THREAD

**Issue by samuelpowell** [JuliaLang/julia#10509]
Building master with the Make.user specified by @Keno for https://github.com/Keno/Cxx.jl (which uses the development version of LLVM) gives the following error:

```
disasm.cpp: In function ‘void jl_dump_function_asm(uintptr_t, size_t, size_t, const llvm::object::ObjectFile*, llvm::formatted_raw_ostream&)’:
disasm.cpp:343:67: error: no matching function for call to ‘llvm::Target::createMCCodeEmitter(llvm::MCInstrInfo&, llvm::MCRegisterInfo&, llvm::MCSubtargetInfo&, llvm::MCContext&) const’
         CE = TheTarget->createMCCodeEmitter(*MCII, *MRI, *STI, Ctx);
                                                                   ^
disasm.cpp:343:67: note: candidate is:
In file included from disasm.cpp:58:0:
/scratch/juliacxx/usr/include/llvm/Support/TargetRegistry.h:406:20: note: llvm::MCCodeEmitter* llvm::Target::createMCCodeEmitter(const llvm::MCInstrInfo&, const llvm::MCRegisterInfo&, llvm::MCContext&) const
     MCCodeEmitter *createMCCodeEmitter(const MCInstrInfo &II,
                    ^
/scratch/juliacxx/usr/include/llvm/Support/TargetRegistry.h:406:20: note:   candidate expects 3 arguments, 4 provided
``` 

This is due to r231861 in LLVM. This can easily be fixed by changing the problematic call to

```
CE = TheTarget->createMCCodeEmitter(*MCII, *MRI, Ctx);
````

which permits the build and all tests to complete successfully.

I have not submitted a pull request since I am unfamiliar with the Julia codebase, and unsure if you are accepting fixes related to the development branch of LLVM.

**Comment by nalimilan** [comment]
Yes, if you have a fix, I think it will be accepted if it's within `#ifdef`s so that the build also works with previous LLVM versions.

**Comment by samuelpowell** [comment]
Okay, I'll submit a pull request. Thanks.

**Comment by samuelpowell** [comment]
Fixed in #10511
